### PR TITLE
Improve the Spanner Features Sample and Docs

### PR DESCRIPTION
**Pull request by dzou**
Improve the samples and documentation which show how to access the JDBC driver features from Hibernate.

### CODE REVIEW

File: google-cloud-spanner-hibernate-samples/basic-spanner-features-sample/README.adoc [hunk 1]
```diff
@@ -15,18 +15,53 @@ Alternatively, have the https://cloud.google.com/sdk/[Google Cloud SDK] installe
...
+```
+jdbc:cloudspanner:/projects/{YOUR_PROJECT_ID}/instances/{YOUR_INSTANCE_ID}/databases/{YOUR_DATABASE_ID}
+```
+
+=== Stale Read Sample
+
+Stale reads are a feature in Cloud Spanner which allows you to read records from the database https://cloud.google.com/spanner/docs/reads#read_types[at a previous point in time].
+
+The `StaleReadsDemo.java` provides a demonstration of how to perform a Stale read in Hibernate.
+The sample creates a new row in the database which is successfully read by a strong-read, but it is not present in the stale read.
```
Comment (elefeint) [inline google-cloud-spanner-hibernate-samples/basic-spanner-features-sample/README.adoc:32]:
```suggestion
The sample creates a new row in the database which is successfully read by a strong-read, but is not present in the stale read.
```

File: google-cloud-spanner-hibernate-samples/basic-spanner-features-sample/README.adoc [hunk 2]
```diff
@@ -15,18 +15,53 @@ Alternatively, have the https://cloud.google.com/sdk/[Google Cloud SDK] installe
...
+Many advanced features of Cloud Spanner can be accessed through the JDBC Driver by unwrapping the underlying JDBC connection from the session:
+
+```
+session.doWork(conn -> {
+  conn.setReadOnly(false);
+  conn.createStatement().execute("SET READ_ONLY_STALENESS = 'STRONG'");
+});
+```
+
+NOTE: It is important to remember that JDBC Connections are pooled and **shared** among Hibernate sessions. If you modify a JDBC connection in one session (such as setting it to read-only), that change will affect the next session using the connection. Therefore, it is recommended to wrap session creation in a helper class (such as link:src/main/java/com/example/SessionHelper.java[`SessionHelper.java`] in this sample) to help manage accessing JDBC driver settings.
```
Comment (elefeint) [inline google-cloud-spanner-hibernate-samples/basic-spanner-features-sample/README.adoc:63]: Just "to help manage" maybe?

File: google-cloud-spanner-hibernate-samples/basic-spanner-features-sample/src/main/java/com/example/SessionHelper.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  }
+
+  /**
+   * Creates a read-write session.
+   */
+  public Session createReadWriteSession() {
+    Session session = sessionFactory.openSession();
+    session.doWork(conn -> {
+      conn.setReadOnly(false);
+      conn.createStatement().execute("SET READ_ONLY_STALENESS = 'STRONG'");
```
Comment (elefeint) [inline google-cloud-spanner-hibernate-samples/basic-spanner-features-sample/src/main/java/com/example/SessionHelper.java:48]: Leave a comment about this not being a mistake. What does happen if a read/write session has staleness settings set? I thought read/write settings did not allow anything but "strong consistency as of transaction start"?

**Review by elefeint — APPROVED** [review]

File: google-cloud-spanner-hibernate-samples/basic-spanner-features-sample/README.adoc (hunk 2 above)
Comment (dzou) [reply google-cloud-spanner-hibernate-samples/basic-spanner-features-sample/README.adoc]: Done.

File: google-cloud-spanner-hibernate-samples/basic-spanner-features-sample/src/main/java/com/example/SessionHelper.java (hunk 3 above)
Comment (dzou) [reply google-cloud-spanner-hibernate-samples/basic-spanner-features-sample/src/main/java/com/example/SessionHelper.java]: Done. Ah you're right, good point. Removed.

**Pull request merged**